interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -432,32 +441,42 @@ MathFuncts, errGen) {
         //      comprised of `signatureObject`s by walking through the
         //      `firstSampleSignatures` and constructing a `signatureObject` for
         //      each. [outer .map()]
-        var signatureSet = firstSampleSignatures.map(function(signatureId) {
-          // build the raw signatureSet
-          var mapSampleIdsToActivity = function(sampleId) {
-            // (2b) the array of activity for each signature is built by
-            //      plucking the activity `.value` for each sample within this
-            //      signature from the `activityCache` [inner .map()]
-            // FIXME: counting on array order to match signature order here
-            return cbSampleBin.activityCache.get(sampleId)[signatureId - 1]
-              .value;
-          };
-          var signatureObject = {
-            'id': signatureId,
-            'name': cbSampleBin.getCachedSignature(signatureId).name,
-            'activityA': sg['base-group'].map(mapSampleIdsToActivity),
-            'activityB': sg['comp-group'].map(mapSampleIdsToActivity)
-          };
-          signatureObject.diff = (
-            MathFuncts.mean(signatureObject.activityA) -
-            MathFuncts.mean(signatureObject.activityB)
-          );
-          signatureObject.rawPValue = MathFuncts.tTest(
-            signatureObject.activityA, signatureObject.activityB
-          ).pValue();
-
-          return signatureObject;
-        });
+        var signatureSet = firstSampleSignatures.map(
+          function(signatureId, index) {
+            // build the raw signatureSet
+            var mapSampleIdsToActivity = function(sampleId) {
+              // (2b) the array of activity for each signature is built by
+              //      plucking the activity `.value` for each sample within the
+              //      `index`th signature from the `activityCache`
+              //      [inner .map()]
+              var cachedActivity = cbSampleBin.activityCache.get(sampleId);
+              if (cachedActivity[index].node !== signatureId) {
+                // ensure we're pulling out the right node
+                $log.error(
+                  'mapSignaturesToSignatureInfo: signature IDs do not match.' +
+                  ' First sample = ' + signatureId + ', but sample ' +
+                  sampleId + ' =', cachedActivity[index]
```

Comment:
This case is slightly less verbose (note that the "first sample," in this case, is including just the `signatureId`. I would have done this the same way as the other error for the same reasons, but we are mapping an array that has already been translated from objects to IDs, so I no longer have the full object.

Anyway, let me know if you still think this is too much to include in the log, given that I expect this will be a rare error condition.